spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -914,6 +916,63 @@ def run_script(self, filename, wdir, args, debug, post_mortem,
                   "<br><br>Please open a new one and try again."
                   ) % osp.basename(filename), QMessageBox.Ok)
 
+    def run_cell(self, code, cell_name, filename):
+        """Run cell in current or dedicated client"""
+
+        def norm(text):
+            return remove_backslashes(to_text_string(text))
+
+        self.run_cell_filename = filename
+
+        # Select client to execute code on it
+        client = self.get_client_for_file(filename)
+        if client is None:
+            client = self.get_current_client()
+
+        is_internal_kernal = False

```

Comment:
`is_internal_kernal` -> is_internal_kern*e*l`

Reply:
Also, we need to change this a little bit. `runcell` will also work in external *Spyder* kernels (i.e. kernels started with `python -m spyder_kernels.console`).

So we need to add a method to `client` that detects if kernels are of Spyder kind or not, and then avoid using `runcell` the code if they are not.

Reply:
I think this might be best left for another PR (I can make the issue) since this is true for the `run_script` function above using `runfile()` instead of `%run` on external consoles as well. The changes might as well be made at the some time.

I guess currently the `run_script` function wouldn't work too well on a remote machine unless the local files were mirrored in the remote machine. If this is the case one solution would be to run the contents of the whole file through runcell. Of course having real remote file editing would be better, but in the near term it could make remote kernels more friendly.

Reply:
> I think this might be best left for another PR (I can make the issue)

Ok, no problem. Please open the issue then, I'll take care of it.